provisioning/device/src/transport_state_machine.ts

```diff
@@ -0,0 +1,282 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+'use strict';
+
+import { EventEmitter } from 'events';
+import { errors, X509, SharedAccessSignature } from 'azure-iot-common';
+import * as machina from 'machina';
+import * as Provisioning from './transport_interface';
+import * as dbg from 'debug';
+const debug = dbg('azure-device-provisioning:transport-fsm');
+
+export interface TransportHandlers {
+  _doConnectForFsm(callback: (err?: Error) => void): void;
+  _doDisconnectForFsm(callback: (err?: Error) => void): void;
+  _doFirstRegistrationRequestForFsm(registrationId: string, authorization: SharedAccessSignature | X509 | string, requestBody: any, forceRegistration: boolean, callback: (err?: Error, responseBody?: any, result?: any, pollingInterval?: number) => void): void;
+  _doOperationStatusQueryForFsm(registrationId: string, operationId: string, callback: (err?: Error, responseBody?: any, result?: any, pollingInterval?: number) => void): void;
+  _getErrorFromResultForFsm(result: any): any;
```

Comment:
That is an very weird interface. by convention anything prefixed by underscore is private in javascript. in typescript you don't need to do that if you use the `private` access modifier. not using it makes me think you want these to be public. so they shouldn't be prefixed. 
Also the `forFsm` suffix suggest strong coupling between the FSM and the transport. I'm not sure if there is such a strong coupling... and think we could dispense with the Suffix.

Reply:
Suggesting: 
```js
export interface TransportHandlers {
  connect(callback: (err?: Error) => void): void;
  disconnect(callback: (err?: Error) => void): void;
  registrationRequest(registrationId: string, authorization: SharedAccessSignature | X509 | string, requestBody: any, forceRegistration: boolean, callback: (err?: Error, responseBody?: any, result?: any, pollingInterval?: number) => void): void;
  queryRegistrationStatus(registrationId: string, operationId: string, callback: (err?: Error, responseBody?: any, result?: any, pollingInterval?: number) => void): void;
  getErrorResult(result: any): any;
}
```

---
In reply to: [146399275](https://github.com/Azure/azure-iot-sdk-node/pull/115#discussion_r146399275) [](ancestors = 146399275)